### PR TITLE
Refactor empty state handling across tables; implement tests

### PR DESCRIPTION
**Pull request by gkarat**
Implements https://issues.redhat.com/browse/CCXDEV-8280, https://issues.redhat.com/browse/CCXDEV-8242

Mainly refactors how table components are handling empty states. Also adds tests to check that the empty state titles, messages, and icons are rendered correctly.

- [x] Test "no match" empty states in each table
- [x] Check headers always displayed
- [x] Check icons are not rendered (only title and text body): https://marvelapp.com/prototype/g70a1he/screen/82026901

Some illustrations:

Screenshot 2022-06-03 at 15-53-02 37556893-a47d-4b14-8e8b-e0f741f7df4f - Clusters - OCP Advisor Red Hat Insights
Screenshot 2022-06-03 at 15-52-30 No Issues Cluster - Clusters - OCP Advisor Red Hat Insights
Screenshot 2022-06-03 at 15-52-17 Test disable recommendation - Clusters - OCP Advisor Red Hat Insights
Screenshot 2022-06-03 at 15-51-59 Recommendations - OCP Advisor Red Hat Insights
Screenshot 2022-06-03 at 15-51-45 Data loss will happen when a machine object is deleted that maps to a 'NotReady' node with attached VMDK volumes - Recommendations - OCP Advisor Red Hat Insights

### CODE REVIEW

**Comment by gkarat** [comment]
@RedHatInsights/team-interact, @ikerreyes, please take a look 👋🏼

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js [hunk 1]
```diff
@@ -691,8 +703,8 @@ describe('empty successful affected clusters table', () => {
   });
 
   it('renders no clusters message', () => {
-    cy.get('#empty-state-message')
-      .find('h4')
+    cy.get('[ouiaid="empty-state"]')
```
Comment (ikerreyes) [inline src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:706]: cannot we use `checkEmptyState` with other title?

File: cypress/utils/table.js [hunk 2]
```diff
@@ -33,9 +33,24 @@ function tableIsSortedBy(columnTitle) {
...
+  cy.get(TABLE)
+    .find('[ouiaid=empty-state]')
+    .should('have.length', 1)
+    .within(() => {
+      cy.get('.pf-c-empty-state__icon').should(
+        'have.length',
+        checkIcon ? 1 : 0
+      );
+      cy.get(`h5${TITLE}`).should('have.text', title);
+      cy.get('.pf-c-empty-state__body').should('have.text', body);
```
Comment (ikerreyes) [inline cypress/utils/table.js:46]: Maybe just checking the title is enough. If anytime the body text changes, we do not need to update anything. But it is also fine to test for it.

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js [hunk 3]
```diff
@@ -477,8 +480,11 @@ describe('non-empty successful affected clusters table', () => {
             let sortedNames = _.map(filterData(filters), 'name');
             filterApply(filters);
             if (sortedNames.length === 0) {
-              // TODO check empty table view
-              // TODO headers are displayed
+              checkEmptyState(
```
Comment (ikerreyes) [inline src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:483]: Given that we use the same title an body in multiple places, should we create a partial function for it?

File: src/Components/Cluster/Cluster.spec.ct.js [hunk 4]
```diff
@@ -62,7 +63,10 @@ describe('cluster page', () => {
     // renders table component
     cy.get(RULES_TABLE).should('have.length', 1);
     // test how many rows were rendered
-    checkRowCounts(singleClusterPageReport.report.data.length);
+    // TODO: for selectable rows, tbody is rendered for each row, and checkRowCounts doesn't work correctly
```
Comment (ikerreyes) [inline src/Components/Cluster/Cluster.spec.ct.js:66]: maybe then we should improve the checkRowCounts function rather than creating a ad-hoc test

File: src/Components/Cluster/Cluster.spec.ct.js [hunk 5]
```diff
@@ -90,7 +94,10 @@ describe('cluster page', () => {
     cy.get(CLUSTER_HEADER).should('have.length', 1);
     // renders table component
     cy.get(RULES_TABLE).should('have.length', 1);
-    cy.get('#loading-skeleton').should('have.length', 1);
+    cy.get('[data-ouia-component-id=loading-skeleton]').should(
```
Comment (ikerreyes) [inline src/Components/Cluster/Cluster.spec.ct.js:97]: there is function for this: `cy.ouiaId('loading-skeleton').should(...`

File: src/Components/Cluster/Cluster.spec.ct.js [hunk 6]
```diff
@@ -138,7 +145,11 @@ describe('cluster page', () => {
     cy.get(FILTER_CHIPS).each(($el) =>
       expect($el.text()).to.be.oneOf(['foo bar', 'Low', 'Performance'])
     );
-    checkRowCounts(0);
+    checkRowCounts(1);
+    checkEmptyState(
```
Comment (ikerreyes) [inline src/Components/Cluster/Cluster.spec.ct.js:149]: Maybe then check empty state should also check that row count is 1?

File: src/Components/RecsListTable/RecsListTable.spec.ct.js [hunk 7]
```diff
@@ -783,9 +797,11 @@ describe('empty recommendations list table', () => {
   });
 
   it('renders error message', () => {
-    cy.get('#error-state-message')
-      .find('h4')
-      .should('have.text', 'Something went wrong');
+    checkEmptyState(
```
Comment (ikerreyes) [inline src/Components/RecsListTable/RecsListTable.spec.ct.js:800]: This is not empty state, but  a message, right? Should we rename the method to `checkMessage`?

**Review by ikerreyes — COMMENTED** [review]
Nice, thanks for the work.

Just left some minor comments.

File: cypress/utils/table.js (hunk 2 above)
Comment (gkarat) [reply cypress/utils/table.js]: this make sense :)

File: src/Components/RecsListTable/RecsListTable.spec.ct.js (hunk 7 above)
Comment (ikerreyes) [reply src/Components/RecsListTable/RecsListTable.spec.ct.js]: So in this test, the `something wen wrong` is displayed withing an EmptyState object?

Comment (ikerreyes) [reply src/Components/RecsListTable/RecsListTable.spec.ct.js]: ok
